packages/vaadin-grid/test/keyboard-navigation.test.js

```diff
@@ -1956,17 +1956,42 @@ describe('keyboard navigation', () => {
 
       const spy = sinon.spy();
 
-      grid.addEventListener('cell-focus', (e) => {
+      const eventHandler = (e) => {
         spy();
 
         const context = e.target.getEventContext(e);
         expect(context).to.deep.equal(expectedContext);
         done();
-      });
+      };
+
+      grid.addEventListener('cell-focus', eventHandler);
+
+      left();
+
+      expect(spy.calledOnce).to.be.true;
+      grid.removeEventListener('cell-focus', eventHandler);
+    });
+
+    it('should contain event context in event detail', (done) => {
+      tabToBody();
+      right();
+
+      const spy = sinon.spy();
+
+      const eventHandler = (e) => {
+        spy();
+
+        expect(e.detail).not.to.be.empty;
+        done();
+      };
+
+      grid.addEventListener('cell-focus', eventHandler);
 
       left();
 
       expect(spy.calledOnce).to.be.true;
+
+      grid.removeEventListener('cell-focus', eventHandler);
```

Comment:
You probably don't need to remove event listeners manually since they are automatically cleaned up as soon as  `fixtureSync` [cleans up](https://github.com/vaadin/testing-helpers/blob/master/src/fixture.ts#L13) created fixtures.

Reply:
I think the issue with this test suite is that the fixture is set up once for all tests, by using `before` instead of `beforeEach`. There are many other tests in this suite that reset the DOM in some way.

Reply:
True, didn't notice that.